R/tokenizers.R

```diff
@@ -39,60 +39,35 @@ NULL
 #'   stri_replace_all_regex stri_detect_fixed stri_replace_all_fixed
 #' @export
 tokenize_word <- function(x, split_hyphens = FALSE, verbose = quanteda_options("verbose")) {
-    named <- names(x)
+    
+    m <- names(x)
+    x[is.na(x)] <- "" # make NAs ""
 
     # remove variant selector & whitespace with diacritical marks
     x <- stri_replace_all_regex(x, c("[\uFE00-\uFE0F]", "\\s[\u0300-\u036F]"), "",
                                 vectorize_all = FALSE)
     # substitute characters not to split
     x <- preserve_special(x, split_hyphens = split_hyphens, split_tags = TRUE, verbose = verbose)
-    # make NAs ""
-    x[is.na(x)] <- ""
 
     if (verbose) catm("...segmenting tokens\n")
-    structure(stri_split_boundaries(x, type = "word"), names = named)
+    structure(stri_split_boundaries(x, type = "word"), names = m)
 }
 
 #' @rdname tokenize_internal
 #' @importFrom stringi stri_replace_all_regex stri_detect_fixed stri_split_boundaries
 #' @export
 tokenize_word2 <- function(x, split_hyphens = FALSE, verbose = quanteda_options("verbose")) {
```

Comment:
would have made more sense to create a word3 for comparisons to word2...

Reply:
Yes, but the new `word2` is faster than `word` already.

Reply:
Let's swap them back then, and have word2 become word, and word become word1 to indicate it's a legacy tokenizer that is inferior but kept just for backward compatibility.

Reply:
Done in fcc2e6f